src/app/controllers/revGraphView.js

```diff
@@ -0,0 +1,1044 @@
+              });
+              link.classed("link-active", function(o) {
+                return o.source === d || o.target === d ? true : false;
+              });
+              d3.select(this).classed("node-active", true);
+              d3.select(this).select("use").transition() // TODO: fix node size growth
+                .duration(750)
+                .attr("r", (d.nodeScale * 2 + 12) * 1.5);
+            })
+      
+            node.on("mouseout", function(d) {
+              node.classed("node-active", false);
+              link.classed("link-active", false);
+              d3.select(this).select("use").transition() // TODO: fix link size growth
+                .duration(750)
+                .attr("r", d.nodeScale * 2 + 12);
+            });
+
+            scope.sidebarGraphEvent = function(nodeIconId){
+              // Sidebar actions
+              var nodeList = node._groups[0];
+              var iconType = nodeIconId.split(":")[0]
+              var iconId = nodeIconId.split(":")[1]
+              var sideIconId = iconType + "_sideButton_" + iconId;
+              // Hide/show the selected sidebar icon
+              // Store the original opacity of a sidebar button
+              var selectedIcon = d3.select("#" + sideIconId)._groups[0][0]
+              var realIconOpacity = 1
+              // Define the current sidebar icon state
+              var iconState = selectedIcon.iconState ? false : true;
+              var newIconOpacity = iconState ? 0.3 : realIconOpacity;
+              // Hide or show the elements
+              selectedIcon.style.opacity = newIconOpacity;
+              // Update the real icon opacity
+              selectedIcon.origIconOpacity = realIconOpacity;
+              // Update whether or not the elements are active
+              selectedIcon.iconState = iconState;
+
+              // Iterate nodes
+              var nod;
+              for (nod of nodeList) {
+                if (nod.innerHTML.includes('#' + iconType)) {
+                  // Hide/show selected nodes
+                  // Check and store the original opacity of a node
+                  var origNodeOpacity = nod.origNodeOpacity ? false : true;
+                  var realNodeOpacity = origNodeOpacity ? nod.getElementsByTagName('use')[0].style.opacity : nod.origNodeOpacity;
+                  // Determine if current node is visible
+                  var nodActive = nod.active ? false : true;
+                  var newNodeOpacity = nodActive ? 0 : realNodeOpacity;
+                  var nodData = nod.__data__;
+                  if (nodData == undefined) {continue;};
+                  // Hide or show the elements
+                  nod.getElementsByTagName('use')[0].style.opacity = newNodeOpacity;
+                  // Update the real node opacity
+                  nod.origNodeOpacity = realNodeOpacity;
+                  // Update whether or not the elements are active
+                  nod.active = nodActive;
```

Comment:
quite hacky. This feels like you're fighting against Angular/HTML/CSS instead of using the capabilities of the framework. Is there a cleaner way to do this? You already have a place where the opacity of a node is calculated? Try to keep models (e.g. nodes/links/selectedNode/selectedNodeFilter) separate from rendered elements (svg nodes, html elements and attributes).

Reply:
Refactored following these suggestions.